File: leetcode/3001-4000/3201-3300/3221-3230/3223-minimum-length-of-string-after-operations/MinimumLengthOfStringAfterOperations.java
public class MinimumLengthOfStringAfterOperations {
  public static void main(String[] args) {
    System.out.println(new Solution().minimumLength("abaacbcbba"));
  }
}

class Solution {
  public int minimumLength(String s) {
    int[] totalCharacters = new int[26];

    for (int i = 0; i < s.length(); i++) {
      totalCharacters[s.charAt(i) - 'a']++;
    }

    int l = 0;

    for (int i = 0; i < 26; i++) {
      if (totalCharacters[i] > 0) {
        l += (totalCharacters[i] % 2 == 0) ? 2 : 1;
      }
    }

    return l;
  }
}

File: leetcode/3001-4000/3201-3300/3221-3230/3223-minimum-length-of-string-after-operations/MinimumLengthOfStringAfterOperations.ts
function minimumLength(s: string): number {
  const characters = s.split("").reduce(
    (accumulated, value) => {
      accumulated[value] = (accumulated[value] ?? 0) + 1
      return accumulated;
    }, {}
  )
  const totalCharacters = Object.values(characters) as number[];

  return totalCharacters.reduce(
    (accumulated, value) => {
      return accumulated + (value % 2 === 0 ? 2 : 1)
    }, 0
  )
};

console.log(minimumLength("abaacbcbb"));
